jupyter-extension/src/plugins/mount/mount.tsx

```diff
@@ -299,30 +309,48 @@ export class MountPlugin implements IMountPlugin {
     app.shell.add(this._panel, 'left', {rank: 100});
   }
 
+  /**
+   * Checks if the widget currently in focus is a NotebookPanel.
+   * @param widget is an optionally-provided Widget that will be type narrowed.
+   */
+  isCurrentWidgetNotebook = (
+    widget?: Widget | null,
+  ): widget is NotebookPanel => {
+    widget = widget ?? this._widgetTracker.currentWidget;
+    return widget instanceof NotebookPanel;
+  };
+
   getActiveNotebook = (): NotebookPanel | null => {
-    return this._tracker.currentWidget;
+    const currentWidget = this._widgetTracker.currentWidget;
+    return this.isCurrentWidgetNotebook(currentWidget) ? currentWidget : null;
   };
 
   /**
-   * This handles when a notebook is switched to another notebook.
+   * This handles when the focus is switched to another widget.
    * The parameters are automatically passed from the signal when a switch occurs.
    */
-  handleNotebookChanged = async (
-    tracker: INotebookTracker,
-    notebook: NotebookPanel | null,
+  handleWidgetChanged = async (
+    _widgetTracker: ILabShell,
+    widget: ILabShell.IChangedArgs,
   ): Promise<void> => {
-    // Set the current notebook and wait for the session to be ready
-    let context: PpsContext;
-    if (notebook) {
-      await notebook.sessionContext.ready;
-      notebook.context.fileChanged.connect(this.handleNotebookReload);
-      context = {
-        config: this.getNotebookMetadata(notebook),
-        notebookModel: notebook.context.contentsModel,
-      };
-    } else {
-      context = {config: null, notebookModel: null};
+    if (this.isCurrentWidgetNotebook(widget.newValue)) {
+      await this.handleNotebookChanged(widget.newValue);
     }
+    this.setShowPipeline(this._showPipeline);
+    await Promise.resolve();
+  };
+
+  /**
+   * This handles when a notebook is switched to another notebook.
+   */
+  handleNotebookChanged = async (notebook: NotebookPanel): Promise<void> => {
```

Comment:
this function could technically be absorbed into `handleWidgetChanged` as its not called anywhere else.